caom2utils/caom2utils/fits2caom2.py

```diff
@@ -1642,8 +1653,10 @@ def _get_from_list(self, lookup, index, current=None):
         if (keywords and not ObsBlueprint.is_fits(keywords)
                 and not ObsBlueprint.is_function(keywords)):
             value = keywords
-        elif current:
-            value = current
+        elif self._blueprint.update:
+            if (current is not None or
+                    (current is None and isinstance(value, bool))):
```

Comment:
The first clause: the default value for the current parameter in the method signature is 'None', so do not want accidentally assign the default value.

The second clause: boolean attributes are used to represent three different values: True, False, and unknown. For boolean attributes _only_ I assessed that the risk of setting to None accidentally was better than the risk of not setting a value of 'unknown'.

Does that make sense?

Reply:
Now I see your point - I think it would be valuable to document that in the code.  It could be re-written as `if isinstance(value, bool) or current is not None:` to highlight the point.

Reply:
I've added the explanation as a code comment, and re-did the logic as above. Thanks for removing if's and and's!